### PR TITLE
Add basic release workflows for main, stg, and prod acorn images

### PR DESCRIPTION
**Pull request by njhale**
Signed-off-by: Nick Hale <4175918+njhale@users.noreply.github.com>

### CODE REVIEW

File: .github/workflows/release-stg.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: release-stg 
+
+on:
+  workflow_dispatch:
```
Comment (njhale) [inline .github/workflows/release-stg.yaml:4]: Do we want this on a cron too?

File: .github/workflows/release-main.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        with:
+          acorn-version: main
+      - uses: acorn-io/actions-login@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build and Push
+        run: |
+          acorn build --push -t ghcr.io/${{ github.repository }}:main .
```
Comment (njhale) [inline .github/workflows/release-main.yaml:22]: sanity check: we don't need to worry about building for different architectures, do we?

File: .github/workflows/release-main.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+name: release-main 
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build-and-push-main:
+    runs-on: buildjet-4vcpu-ubuntu-2004
```
Comment (njhale) [inline .github/workflows/release-main.yaml:9]: Looking quickly, I couldn't find a great way to test using this particular runner without merging OR subscribing to buildjet using my personal account. I ended up testing against `ubuntu-latest` in my fork.

**Review by njhale — COMMENTED** [review]
There's some redundancy between `release-stg` and `release-prod`, but I figure this is at least clear and makes it easy to diverge if we want to down the line.

File: .github/workflows/release-main.yaml (hunk 3 above)
Comment (thedadams) [reply .github/workflows/release-main.yaml:9]: This is a public repo, so I think this should/can be `ubuntu-latest`.

File: .github/workflows/release-stg.yaml (hunk 1 above)
Comment (thedadams) [reply .github/workflows/release-stg.yaml:4]: I would say yes because all of our other staging workflows run on a cron.

Comment (StrongMonkey) [reply .github/workflows/release-stg.yaml:4]: yeah we need this to run nightly

File: .github/workflows/release-main.yaml (hunk 3 above)
Comment (tylerslaton) [reply .github/workflows/release-main.yaml:9]: +1 to this just being `ubuntu-latest`.

File: .github/workflows/release-main.yaml (hunk 2 above)
Comment (tylerslaton) [reply .github/workflows/release-main.yaml:22]: It depends - how are we going to be using this? Are just going to provision it ourselves and users will hit it or will users provision it for themselves.

**Review by tylerslaton — APPROVED** [review]

File: .github/workflows/release-main.yaml (hunk 2 above)
Comment (cjellick) [reply .github/workflows/release-main.yaml:22]: no. just worry about our own single arch usecase

**Review by cjellick — APPROVED** [review]

**Pull request merged**